### PR TITLE
MongoDB

### LINKED ISSUE #5: Move to MongoDB
TODO: Move to mongodb so this works on other hosters... you dont need to do this in this PR if you want

_Originally posted by @TheColaber in https://github.com/FlagClicked/Backend/pull/2#discussion_r645801318_

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #5

replace replit database w/ mongodb

Tutorials definition:

Tutorials.get(id) <= gets a tutorial from the id. returns null if not found (async)
Tutorials.new(body, author) <= creates a new tutorial from the new id. returns a Tutorial
Tutorials.edit(body, id, user) <= edits a tutorial. the user param will need a User object. id must be a string. will throw an error if the user is not an admin & not the author

### CODE REVIEW

File: src/Tutorials.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const db = require('monk')('localhost/flagclicked')
```
Comment (TheColaber) [inline src/Tutorials.js:1]: fix this when done... localhost/flagclicked

File: src/Tutorials.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const auth = require('./auth.js')
+let tutorials = db.get('tutorials')
+
+tutorials.createIndex('id', { unique: true })
+var Tutorials = {}
+
+Tutorials.get = async function(id) {
```
Comment (TheColaber) [inline src/Tutorials.js:8]: unnneeded async

File: src/Tutorials.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    let tutorial = await Tutorials.get(id)
+    var editor = await auth.getUser(user)
+
+    if (editor.id !== tutorial.author.id && !editor.admin) {
+        throw 'User cannot edit this tutorial!'
+    }
+
+    await tutorials.update({ id }, { $set: { body, "meta.edited.user": editor, "meta.edited.time": Date.now()} })
+
+}
```
Comment (TheColaber) [inline src/Tutorials.js:54]: Might as well put these in "Tutorials" definition

File: src/index.js [hunk 4]
```diff
@@ -3,13 +3,14 @@ const express = require("express");
...
+const Tutorials = require('./Tutorials.js')
 const start = (new Date()).getTime()
 
 /* >>>>> SETTINGS <<<<< */
 
 const port = 3000;
-const frontendURL = "448me.sse.codesandbox.io";
-const backendURL = "hly3v.sse.codesandbox.io";
+const frontendURL = "localhost:3000";
+const backendURL = "localhost:3000";
```
Comment (TheColaber) [inline src/index.js:13]: todo when done

File: src/index.js [hunk 5]
```diff
@@ -59,46 +61,27 @@ app.get('/auth/me', auth.authMiddleware, async (req, res) => {
...
-      },
-      edited: {
-        time: Date.now(),
-        user: req.user
-      }
-    }
-  }
-
-  await db.set(`tutorial-${id}`, Tutorial)
+  let tutorial = Tutorials.new(req.body, req.user)
```
Comment (TheColaber) [inline src/index.js:78]: Doesn't Tutorials.new return a promise?

File: package.json [hunk 6]
```diff
@@ -10,6 +10,7 @@
     "@replit/database": "^2.0.1",
```
Comment (TheColaber) [inline package.json:10]: uninstall replit/database

**Review by TheColaber — CHANGES_REQUESTED** [review]
Epic Job again, I will be cleaning up code and moving stuff to replit (probably) and implement more features

File: src/Tutorials.js (hunk 2 above)
Comment (FunctionalMetatable) [reply src/Tutorials.js:8]: need to add awaits for tutorials.findOne

File: src/index.js (hunk 5 above)
Comment (FunctionalMetatable) [reply src/index.js]: done

File: src/Tutorials.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    return tutorial
+}
+
+Tutorials.new = async function(body, author) {
+    let allTutorials = await tutorials.list()
+    let tutorial = {
+        id: (allTutorials.length + 1).toString(),
+        author: author,
```
Comment (TheColaber) [inline src/Tutorials.js:25]: This is silly but just... author

**Review by TheColaber — APPROVED** [review]
Lets merge it!

File: src/Tutorials.js (hunk 7 above)
Comment (FunctionalMetatable) [reply src/Tutorials.js]: true

File: src/Tutorials.js (hunk 1 above)
Comment (FunctionalMetatable) [reply src/Tutorials.js:1]: isnt that just the mongodb url?

Comment (TheColaber) [reply src/Tutorials.js:1]: oh it is? whoops

**Comment by FunctionalMetatable** [comment]
could you implement https://github.com/FunctionalMetatable/Backend/commit/7502d23bbbee782960d46effa7464b7a110d7bc1? i pushed it but 1 second before this got merged. tutorials.list does not work, but tutorials.find does.

**Pull request merged**